src/test/javascript/spec/component/shared/metis/posting-content/posting-content-part.component.spec.ts

```diff
@@ -68,9 +97,238 @@ describe('PostingContentPartComponent', () => {
+
+            // should display relevant icon for exercise
+            const icon = getElement(debugElement, 'fa-icon');
+            expect(icon).not.toBe(null);
+            expect(icon.innerHTML).toInclude('fa fa-paperclip');
+
+            // on click should navigate to referenced exercise within current tab
+            referenceLink.click();
+            expect(navigateByUrlSpy).toHaveBeenCalledTimes(1);
+            expect(openAttachmentSpy).toBeCalledTimes(0);
+        });
+
+        it('should contain a reference to lecture with icon', () => {
+            const referenceStr = 'Test Lecture';
+            component.postingContentPart = {
+                contentBeforeReference,
+                linkToReference: ['/courses/1/lectures/1/'],
+                referenceStr,
+                referenceType: ReferenceType.LECTURE,
+                contentAfterReference,
+            } as PostingContentPart;
+            fixture.detectChanges();
+
+            // should display lecture name to user
+            const referenceLink = getElement(debugElement, '.reference');
+            expect(referenceLink).not.toBe(null);
+            expect(referenceLink.innerHTML).toInclude(referenceStr);
+
+            // should display relevant icon for lecture
+            const icon = getElement(debugElement, 'fa-icon');
+            expect(icon).not.toBe(null);
+            expect(icon.innerHTML).toInclude('fa fa-chalkboard-user');
+
+            // on click should navigate to referenced lecture within current tab
+            referenceLink.click();
+            expect(navigateByUrlSpy).toHaveBeenCalledTimes(1);
+            expect(openAttachmentSpy).toBeCalledTimes(0);
+        });
+
+        it('should contain a reference to attachment with icon', () => {
+            const referenceStr = 'Lecture 1 - Slide';
+            const attachmentURL = '/api/files/attachments/lecture/1/Lecture-1.pdf';
+            component.postingContentPart = {
+                contentBeforeReference,
+                referenceStr,
+                referenceType: ReferenceType.ATTACHMENT,
+                attachmentToReference: attachmentURL,
+                contentAfterReference,
+            } as PostingContentPart;
+            fixture.detectChanges();
+
+            // should display file name to user
+            const referenceLink = getElement(debugElement, '.reference');
+            expect(referenceLink).not.toBe(null);
+            expect(referenceLink.innerHTML).toInclude(referenceStr);
+
+            // should display relevant icon for attachment
+            const icon = getElement(debugElement, 'fa-icon');
+            expect(icon).not.toBe(null);
+            expect(icon.innerHTML).toInclude('fa fa-file');
+
+            // on click should open referenced attachment within new tab
+            referenceLink.click();
+            expect(openAttachmentSpy).toBeCalledTimes(1);
+            expect(openAttachmentSpy).toBeCalledWith(attachmentURL);
```

Comment:
Please replace those tests with parameterized tests (this should be done really easy)

Reply:
Useful concept, thanks!